### PR TITLE
Add example on how to wrap and customize the block editor

### PR DESCRIPTION
**Pull request by skogsmaskin**
A working example on how to wrap and customise inputs would be nice to have in the example studio right?

### CODE REVIEW

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  static propTypes = {
+    type: PropTypes.shape({
+      title: PropTypes.string
+    }).isRequired,
+    level: PropTypes.number,
+    value: PropTypes.array,
+    onChange: PropTypes.func.isRequired
+  };
+
+  constructor(props) {
```
Comment (bjoerge) [inline packages/example-studio/components/FunkyEditor/FunkyEditor.js]: Remove the constructor?

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    console.log(props)
+    super(props)
+  }
+
+  handleChange = event => {
+    this.props.onChange(event)
```
Comment (bjoerge) [inline packages/example-studio/components/FunkyEditor/FunkyEditor.js]: If all handleChange do is just call the given `onChange`-prop, maybe just pass it from render()?
```js
render() {
  const {type, value, level, onChange} = this.props
  return (
    //...
    <SlateEditor
      //...
      onChange={onChange}
     />
    //...
  )
}
```

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import SlateEditor from '@sanity/form-builder/lib/inputs/BlockEditor-slate'
```
Comment (bjoerge) [inline packages/example-studio/components/FunkyEditor/FunkyEditor.js]: I'm a bit sceptical to documenting this as the way to get the default block editor, as it makes refactoring the internals of the form-builder package a breaking change. IMO our official stance should be to never recommend importing anything from `./lib`

Comment (bjoerge) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: Just realized that the BlockEditor is exported by default as `SlateInput` https://github.com/sanity-io/sanity/blob/54b0c2209cde2f9e8a70fbe1fe35f9530587b18d/packages/@sanity/form-builder/src/index.js#L12

If we change the exported name to BlockEditor, the import would be:
```js
import {BlockEditor} from '@sanity/form-builder/lib/inputs/BlockEditor-slate'
```

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    type: PropTypes.shape({
+      title: PropTypes.string
+    }).isRequired,
+    level: PropTypes.number,
+    value: PropTypes.array,
+    onChange: PropTypes.func.isRequired
+  };
+
+  handleChange = event => {
+    this.props.onChange(event)
```
Comment (bjoerge) [inline packages/example-studio/components/FunkyEditor/FunkyEditor.js]: If all `handleChange` do is just call the given `onChange`-prop, maybe just pass it from render()?
```js
render() {
  const {type, value, level, onChange} = this.props
  return (
    //...
    <SlateEditor
      //...
      onChange={onChange}
     />
    //...
  )
}
```

**Comment by bjoerge** [comment]
This is nice! Possible future work could be to turn it into an example of how to use it with an external service like readable.io.

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js (hunk 3 above)
Comment (skogsmaskin) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: That would be nice yes!

Comment (bjoerge) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: Renamed in #210

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js (hunk 4 above)
Comment (skogsmaskin) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: I think it has some value to show that you can make your own handler and do stuff. As you say in the below comment (readable.io).

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js (hunk 3 above)
Comment (skogsmaskin) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: I just realized that @sanity/form-builder is not a direct dependency in the studio :(

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js (hunk 4 above)
Comment (bjoerge) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: True, but then maybe we should add an example where we are actually doing something with the value (or at least a comment in `handleChange`describing that we _could_ do something with the value if we wanted to)

File: packages/example-studio/components/FunkyEditor/FunkyEditor.js (hunk 3 above)
Comment (bjoerge) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: Fixed in #210 by exporting it from the sanity form builder part

Comment (bjoerge) [reply packages/example-studio/components/FunkyEditor/FunkyEditor.js]: So the import would now be: ```js
import {BlockEditor} from 'part:@sanity/form-builder'
```

**Comment by bjoerge** [comment]
💖

**Pull request merged**